sunpy/coordinates/transformations.py

```diff
@@ -136,6 +141,21 @@ def transform_with_sun_center():
         _ignore_sun_motion = old_ignore_sun_motion
 
 
+@contextmanager
+def propagate_with_solar_surface():
+    with transform_with_sun_center():
+        try:
+            global _apply_diffrot
+
+            old_apply_diffrot = _apply_diffrot  # nominally False
+
+            log.debug("Apply solar differential rotation for any changes in obstime")
+            _apply_diffrot = True
+            yield
+        finally:
+            _apply_diffrot = old_apply_diffrot
```

Comment:
Add another debug here so you know when it turns off? Might make the debug output a little easier to parse?

Reply:
Good suggestion, and the same for `transform_with_sun_center()`.